File: Programmers/week4/64061/jjunii/64061.py
from collections import deque
    
def solution(board, moves):
    n = len(board)
    basket = []     #stack
    
    board_list = [deque([]) for _ in range(n)]  # idx: 크레인 위치, value: 인형 배열
    for r in range(n):
        for c in range(n):
            if board[r][c] == 0:
                continue
            board_list[c].append(board[r][c])
    
    cnt = 0
    for m in [m - 1 for m in moves]:    # 1-based -> 0-based
        if not board_list[m]:
            continue
            
        item = board_list[m].popleft()
        if basket and basket[-1] == item:
            basket.pop()
            cnt += 2
        else:
            basket.append(item)
    
    return cnt

File: Programmers/week4/64061/kwonjin2/64061.js
function solution(board, moves) {
    const stack = [];
    let result = 0;
    
    for (let move of moves) {
        const col = move - 1;
        
        for (let row = 0; row < board.length; row++) {
            const doll = board[row][col];
            
            if (doll !== 0) {
                board[row][col] = 0;
                
                if (stack[stack.length - 1] === doll) {
                    stack.pop();
                    result += 2;
                } else {
                    stack.push(doll);
                }
                
                break;
            }
        }
    }
    
    return result;
}